WireDomain/Sources/WireDomain/Repositories/Conversations/ConversationLocalStore.swift

```diff
@@ -0,0 +1,399 @@
+
+    func deleteConversation(
+        withQualifiedId qualifiedId: WireAPI.QualifiedID
+    ) async throws
+
+    /// Removes a SelfUser from the specified conversation.
+    /// - Parameter qualifiedId: The conversation qualified ID.
+
+    func removeSelfUserFromConversation(
+        withQualifiedId qualifiedId: WireAPI.QualifiedID
+    ) async
+
+    /// Stores a flag indicating whether a conversation requires update from backend.
+    /// - Parameter qualifiedId: The conversation qualified ID.
+
+    func storeConversationNeedsBackendUpdate(
+        _ needsUpdate: Bool,
+        qualifiedId: WireAPI.QualifiedID
+    ) async
+
+    /// Stores the specified failed conversation
+    /// - Parameter qualifiedId: The conversation qualified ID.
+
+    func storeFailedConversation(
+        withQualifiedId qualifiedId: WireAPI.QualifiedID
+    ) async
+}
+
+public final class ConversationLocalStore: ConversationLocalStoreProtocol {
+
+    // MARK: - Properties
+
+    let context: NSManagedObjectContext
+
+    // MARK: - Object lifecycle
+
+    public init(
+        context: NSManagedObjectContext
+    ) {
+        self.context = context
+    }
+
+    // MARK: - Public
+
+    public func storeConversation(
+        _ conversation: WireAPI.Conversation,
+        isFederationEnabled: Bool
+    ) async throws {
+        guard let id = conversation.id ?? conversation.qualifiedID?.uuid else {
+            return
+        }
+
+        switch conversation.type {
+        case .group:
+            await updateOrCreateGroupConversation(
+                remoteConversation: conversation,
+                remoteConversationID: id,
+                isFederationEnabled: isFederationEnabled
+            )
+
+        case .`self`:
+            try await updateOrCreateSelfConversation(
+                remoteConversation: conversation,
+                remoteConversationID: id,
+                isFederationEnabled: isFederationEnabled
+            )
+
+        case .connection:
+            /// Conversations are of type `connection` while the connection
+            /// is pending.
+            await updateOrCreateConnectionConversation(
+                remoteConversation: conversation,
+                remoteConversationID: id,
+                isFederationEnabled: isFederationEnabled
+            )
+
+        case .oneOnOne:
+            /// Conversations are of type `oneOnOne` when the connection
+            /// is accepted.
+            await updateOrCreateOneToOneConversation(
+                remoteConversation: conversation,
+                remoteConversationID: id,
+                isFederationEnabled: isFederationEnabled
+            )
+
+        default:
+            /// conversation type is nil
+            return
+        }
+    }
+
+    public func deleteConversation(
+        withQualifiedId qualifiedId: WireAPI.QualifiedID
+    ) async throws {
+        let conversation = await context.perform { [context] in
```

Comment:
I try to stick at best to what current legacy code is currently doing to avoid introducing breaking changes since these repo will soon be consumed (from my understanding) by existing code. So according to legacy code (`ConversationRequestStrategy` class), this only concerned `group` conversations however I can rename it `deleteConversationGroup` to be more explicit.

Reply:
resolving this comment, as per John's comment, as we're dealing with slow sync in this PR, we don't need these methods for now, so I removed them